### DISCUSSION TITLE
Renovate not filtering out packages for existing PRs when new packageRules config added to renovate.json

### DISCUSSION THREAD

**Discussion by Mknight492** [renovatebot/renovate#21807] [Suggest an Idea]
### Type of discussion.

I just want to chat

### Tell us more.

I recently came across an issue where I configured Renovate to make all PR with all "minor" and "patch" updates.

This worked as expected, however I also then wanted to filter the PR to only packages which have a major version >1.0.0 so that they were lower risk package updates.

This works as expected for **new PR** but I don't believe it's working work **existing PRs** event though the PR summary states it's filtered out the non major version >1.0.0 version.

There's a [demonstration repo here](https://github.com/Mknight492/example).

I first configured renovate to merge all minor and patch packages like so:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabled": true,
  "packageRules": [
    {
      "groupName": "All non-major go dependencies",
      "matchUpdateTypes": ["minor", "patch"],
    }
  ]
}
```

Which raised the correct PR updating the packages like so:

github.com/aws/aws-sdk-go-v2 v1.16.1 -> github.com/aws/aws-sdk-go-v2 v1.18.0
github.com/satori/go.uuid v1.1.0 -> github.com/satori/go.uuid v1.2.0
**golang.org/x/tools v0.7.0 -> golang.org/x/tools v0.8.0**

However after updating the renovate.json to exclude the package with an unstable API:

``` json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabled": true,
  "packageRules": [
    {
      "groupName": "All non-major go dependencies",
      "matchUpdateTypes": ["minor", "patch"],
      "matchCurrentVersion": ">=1.0.0"
    }
  ]
}
```

the PR claims that it's no-longer updating the golang.org/x/tools package 

Yet it's still include in the [PR here](https://github.com/Mknight492/example/pull/2/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R8)

Is this intended behavior?
![image](https://user-images.githubusercontent.com/39689669/234425846-2d2a6ce1-1232-4930-8c88-465a818a03b8.png)

Is seems as though it's misleading in the PR summary.

**Comment by rarkins** [comment]
Does it fix itself if you tick the rebase/retry checkbox?

**Reply by Mknight492** [reply]
Yes - rebasing or reopening the issue fixes it.

**Reply by rarkins** [reply]
The issue appears to be this:
- The next time Renovate runs after you modify config, it doesn't "remember" the old config or old PR
- It learns that it needs to have a PR updating two packages, and that such a PR appears to already exist (matching branch name + title)
- It checks if that branch has the required versions it's after - it does
- It (incorrectly) assumes nothing should be done with the branch

The next time it needs to rebase that branch, the problem is fixed.

This can be treated as a bug. Generically, Renovate may not detect when the number of updates in a branch shrinks (it doesn't necessarily need to have anything to do with packageRules)

**Reply by Mknight492** [reply]
Thanks @rarkins for your very quick response and that makes sense to me. I've [raised a Bug here ](https://github.com/renovatebot/renovate/issues/21826) then. I'll wait till it gets reviewed but I'm guessing it wouldn't be a good candidate for first time contributors?

**Reply by rarkins** [reply]
You are right, it's likely a tricky one to fix
